src/goto-programs/remove_exceptions.cpp

```diff
@@ -302,22 +277,32 @@ void remove_exceptionst::instrument_throw(
     {
       goto_programt::targett new_state_pc=
         stack_catch[i][j].second;
+      if(stack_catch[i][j].first==irep_idt())
+      {
+        // Universal handler. Highest on the stack takes
+        // precedence, so overwrite any we've already seen:
+        default_target=new_state_pc;
+      }
+      else
+      {
+        // Normal exception handler, make an instanceof check.
+        goto_programt::targett t_exc=goto_program.insert_after(instr_it);
+        t_exc->make_goto(new_state_pc);
+        t_exc->source_location=instr_it->source_location;
+        t_exc->function=instr_it->function;
 
-      // find handler
-      goto_programt::targett t_exc=goto_program.insert_after(instr_it);
-      t_exc->make_goto(new_state_pc);
-      t_exc->source_location=instr_it->source_location;
-      t_exc->function=instr_it->function;
-
-      // use instanceof to check that this is the correct handler
-      symbol_typet type(stack_catch[i][j].first);
-      type_exprt expr(type);
+        // use instanceof to check that this is the correct handler
+        symbol_typet type(stack_catch[i][j].first);
+        type_exprt expr(type);
 
-      binary_predicate_exprt check(exc_thrown, ID_java_instanceof, expr);
-      t_exc->guard=check;
+        binary_predicate_exprt check(exc_thrown, ID_java_instanceof, expr);
```

Comment:
Does this imply that `remove_instanceof` must always be run after `remove_exceptions`? This needs to made very explicit in some place, or `remove_exceptions` should trigger a call to `remove_instanceof`.